pkgs/stdenv/linux/default.nix

```diff
@@ -324,7 +326,8 @@ in
 
       extraNativeBuildInputs = [ prevStage.patchelf prevStage.paxctl ] ++
         # Many tarballs come with obsolete config.sub/config.guess that don't recognize aarch64.
-        lib.optional (!localSystem.isx86) prevStage.updateAutotoolsGnuConfigScriptsHook;
+        lib.optional (!localSystem.isx86 || localSystem.libc == "musl")
+        prevStage.updateAutotoolsGnuConfigScriptsHook;
```

Comment:
I can't remember if I asked before but I was wondering if we could just do this on x86 as well? Just for consistency.

Reply:
@matthewbauer I would always do it on all platforms, but that's a mass rebuild on hydra-cached platforms. I'm taking advantage of MUSL's tier-n+1 status to do this right on master :D.